### ISSUE TITLE
[Bug]: OneSignal.Notifications.requestPermission returns a boolean wrapped in array instead of boolean on iOS

### ISSUE THREAD

**Issue by hrastnik** [OneSignal/react-native-onesignal#1726]
### What happened?

Using version react-native-onesignal@5.2.2

The call to `OneSignal.Notifications.requestPermission` returns `[true]` instead of `true`, that is, the actual response is wrapped in an array.

The Typescript definitions suggests it returns a boolean.
  

### Steps to reproduce?

```Markdown
Here's the simple code required to reproduce the issue


import React from "react";
import { Button, View } from "react-native";
import { OneSignal } from "react-native-onesignal";

export function App() {
  return (
    <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
      <Button
        title="Request permissions"
        onPress={async () => {
          const response =
            await OneSignal.Notifications.requestPermission(true);

          console.log("response", response);
        }}
      />
    </View>
  );
}


This logs `response [false]` or `response [true]` depending on the outcome of the prompt.
```

### What did you expect to happen?

The returned value should be a `boolean`

### React Native OneSignal SDK version

5.2.2

### Which platform(s) are affected?

- [X] iOS
- [ ] Android

### Relevant log output

```Shell
response [false]
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
@hrastnik thank you for reaching out! We're looking into this.

**Comment by jennantilla** [comment]
@hrastnik thanks for your patience! This should be resolved in [Release 5.2.3](https://github.com/OneSignal/react-native-onesignal/releases/tag/5.2.3). Please let us know if you have any additional questions or concerns!
